custom_components/unifiprotect/camera.py

```diff
@@ -139,26 +194,59 @@ class UnifiProtectCamera(UnifiProtectEntity, Camera):
 
     def __init__(
         self,
-        upv_object,
-        protect_data,
-        server_info,
-        camera_id,
-        disable_stream,
+        protect: ProtectApiClient,
+        protect_data: UnifiProtectData,
+        camera: UnifiCamera,
+        channel: CameraChannel,
+        is_default: bool,
+        secure: bool,
+        disable_stream: bool,
     ):
         """Initialize an Unifi camera."""
-        super().__init__(upv_object, protect_data, server_info, camera_id, None)
-        self._name = self._device_data["name"]
+        super().__init__(protect, protect_data, camera, None)
+
+        self.device: UnifiCamera = camera
+        self.channel = channel
+        self._secure = secure
         self._disable_stream = disable_stream
-        self._stream_source = None if disable_stream else self._device_data["rtsp"]
         self._last_image = None
+        self._async_set_stream_source()
+        if self._secure:
+            self._attr_unique_id = (
+                f"{self.device.id}_{self.device.mac}_{self.channel.id}"
+            )
+        else:
+            self._attr_unique_id = (
+                f"{self.device.id}_{self.device.mac}_{self.channel.id}_insecure"
+            )
+        # only the default (first) channel is enabled by default
+        self._attr_entity_registry_enabled_default = is_default and secure
+        if self._secure:
+            self._attr_name = f"{self.device.name} {self.channel.name}"
+        else:
+            self._attr_name = f"{self.device.name} {self.channel.name} Insecure"
+
+    @callback
+    def _async_set_stream_source(self):
+        disable_stream = self._disable_stream
+        if not self.channel.is_rtsp_enabled:
+            disable_stream = False
+
+        rtsp_url = self.channel.rtsp_url
+        if self._secure:
+            rtsp_url = self.channel.rtsps_url
+
+        self._stream_source = None if disable_stream else rtsp_url
         self._attr_supported_features = SUPPORT_STREAM if self._stream_source else 0
 
     @callback
-    def _async_updated_event(self):
-        self._attr_available = (
-            self._device_data["online"] and self.protect_data.last_update_success
-        )
-        self.async_write_ha_state()
+    def _async_update_device_from_protect(self):
+        super()._async_update_device_from_protect()
+        for channel in self.device.channels:
+            if channel.id == self.channel.id:
+                self.channel = channel
+                break
```

Comment:
Can we avoid a linear search here?

Reply:
I will investigate. This is a fixed size list, so as long as the IDs are consistent within the list, it should be safe to just hardcode the ID -> index map.

Reply:
It looks like the ids always match the indicies (0, 1, 2) and the number channels return is always 3 regradless if RTSP is enabled. I will update this to just use the id = index.